### PR TITLE
apm: add 8.12.0 known issue about fleet ingest pipeline bug

### PR DESCRIPTION
**Pull request by carsonip**
Part of https://github.com/elastic/apm-server/issues/12498

### CODE REVIEW

File: docs/en/observability/apm/known-issues.asciidoc [hunk 1]
```diff
@@ -37,6 +37,26 @@ Failed installing package [apm] due to error: [ResponseError: mapper_parsing_exc
...
 A fix was released in 8.11.2: https://github.com/elastic/kibana/pull/171712[elastic/kibana#171712].
 
+*traces-apm@custom ingest pipeline applied to certain data streams unintentionally* +
+_APM Server versions: 8.12.0_ +
+
+// Describe the conditions in which this issue occurs
+If you're using the Elastic APM Server v8.12.0,
+// Describe the behavior of the issue
+a bug has been introduced, so that `traces-apm@custom` ingest pipeline is now additionally applied to data streams `traces-apm.sampled-*`
+and `traces-apm.rum-*`, and applied twice for `traces-apm-*`. This impacts users with a non-empty `traces-apm@custom` ingest pipeline.
```
Comment (colleenmcginnis) [inline docs/en/observability/apm/known-issues.asciidoc:47]:
```suggestion
the `traces-apm@custom` ingest pipeline is now additionally applied to data streams `traces-apm.sampled-*`
and `traces-apm.rum-*`, and applied twice for `traces-apm-*`. This bug impacts users with a non-empty `traces-apm@custom` ingest pipeline.
```

**Review by colleenmcginnis — APPROVED** [review]

File: docs/en/observability/apm/known-issues.asciidoc (hunk 1 above)
Comment (carsonip) [reply docs/en/observability/apm/known-issues.asciidoc]: thanks, I always appreciate these suggestions!

File: docs/en/observability/apm/known-issues.asciidoc [hunk 2]
```diff
@@ -37,6 +37,26 @@ Failed installing package [apm] due to error: [ResponseError: mapper_parsing_exc
...
+*traces-apm@custom ingest pipeline applied to certain data streams unintentionally* +
+_APM Server versions: 8.12.0_ +
+
+// Describe the conditions in which this issue occurs
+If you're using the Elastic APM Server v8.12.0,
+// Describe the behavior of the issue
+the `traces-apm@custom` ingest pipeline is now additionally applied to data streams `traces-apm.sampled-*`
+and `traces-apm.rum-*`, and applied twice for `traces-apm-*`. This bug impacts users with a non-empty `traces-apm@custom` ingest pipeline.
+
+If you rely on this unintended behavior in 8.12.0, please rename your pipeline to `traces-apm.package@custom` to preserve this behavior in 8.12.1.
```
Comment (carsonip) [inline docs/en/observability/apm/known-issues.asciidoc:49]:
```suggestion
If you rely on this unintended behavior in 8.12.0, please rename your pipeline to `traces-apm.package@custom` to preserve this behavior in later versions.
```
^ will this be clearer that it applies to 8.12.1+? or is it already implied?

Comment (simitt) [reply docs/en/observability/apm/known-issues.asciidoc:49]: The fix actually ended up using `.integration` and not `package` as suffix, see https://github.com/elastic/package-spec/pull/704 and https://github.com/elastic/kibana/pull/175448/files.

Comment (carsonip) [reply docs/en/observability/apm/known-issues.asciidoc:49]: good catch! I copied from PR description without noticing it

File: docs/en/observability/apm/known-issues.asciidoc [hunk 3]
```diff
@@ -37,6 +37,26 @@ Failed installing package [apm] due to error: [ResponseError: mapper_parsing_exc
...
+*traces-apm@custom ingest pipeline applied to certain data streams unintentionally* +
+_APM Server versions: 8.12.0_ +
+
+// Describe the conditions in which this issue occurs
+If you're using the Elastic APM Server v8.12.0,
+// Describe the behavior of the issue
+the `traces-apm@custom` ingest pipeline is now additionally applied to data streams `traces-apm.sampled-*`
+and `traces-apm.rum-*`, and applied twice for `traces-apm-*`. This bug impacts users with a non-empty `traces-apm@custom` ingest pipeline.
+
+If you rely on this unintended behavior in 8.12.0, please rename your pipeline to `traces-apm.package@custom` to preserve this behavior in later versions.
```
Comment (carsonip) [inline docs/en/observability/apm/known-issues.asciidoc:49]:
```suggestion
If you rely on this unintended behavior in 8.12.0, please rename your pipeline to `traces-apm.integration@custom` to preserve this behavior in later versions.
```

**Review by bmorelli25 — APPROVED** [review]

**Review by colleenmcginnis — APPROVED** [review]
🐻

**Pull request merged**